File: Brainfuck/deadfish2.cpp
#include <iostream>
#include <string>

int main() {
	std::string program;
	std::cin >> program;
	
	int accumulator = 0;
	
	size_t ip = 0;
	char instruction;
	while(ip < program.size()) {
		instruction = program[ip];
		
		switch(instruction) {
			case 'i':
				accumulator += 1;
				break;
			case 'd':
				accumulator -= 1;
				break;
			case 's':
				accumulator *= accumulator;
				break;
			case 'o':
				std::cout << accumulator << std::endl;
				break;
		}
		
		if(accumulator == 256 || accumulator == -1) {
			accumulator = 0;
		}
		
		ip++;
	}
}

File: Brainfuck/deadfish2.py
program = input()

accumulator = 0

ip = 0
while ip < len(program):
	instruction = program[ip]
	
	if instruction == "i":
		accumulator += 1
	elif instruction == "d":
		accumulator -= 1
	elif instruction == "s":
		accumulator **= 2
	elif instruction == "o":
		print(accumulator)
	
	if accumulator == 256 or accumulator == -1:
		accumulator = 0
	
	ip += 1